comet_core/api_v0.py

```diff
@@ -107,33 +107,36 @@ def assert_fingerprint_syntax(fingerprint):
         raise ValueError("fingerprint invalid: contains invalid characters")
 
 
-def get_and_check_fingerprint():
+def get_and_check_fingerprint(validate_token=True):
     """Reads the fingerprint from a POST request json data if it
     was a POST request, also checks its syntax.
     Also validate the token passed in the request
 
+    Args:
+        validate_token (bool): A parameter to validate the request token or not
     Raises:
         ValueError: if the POST request did not contain json data,
         or if the json data did not contain a fingerprint
 
     Returns:
         str: fingerprint
     """
+    fingerprint = None
```

Comment:
Why setting the fingerprint to `None` here?

We will break out from the POST-part if fingerprint not in request_json by raising ValueError and (although less clear) from the GET part by the KeyError.

Reply:
Pycharm complains that the value might not be initialized. This makes sure whatever happens, there is a proper return value.